src/ast/ddl.rs

```diff
@@ -2002,22 +2002,45 @@ impl fmt::Display for DropBehavior {
 #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
 #[cfg_attr(feature = "visitor", derive(Visit, VisitMut))]
 pub enum UserDefinedTypeRepresentation {
+    /// Composite type: `CREATE TYPE name AS (attributes)`
     Composite {
         attributes: Vec<UserDefinedTypeCompositeAttributeDef>,
     },
+    /// Enum type: `CREATE TYPE name AS ENUM (labels)`
+    ///
     /// Note: this is PostgreSQL-specific. See <https://www.postgresql.org/docs/current/sql-createtype.html>
     Enum { labels: Vec<Ident> },
+    /// Range type: `CREATE TYPE name AS RANGE (options)`
+    Range {
+        options: Vec<UserDefinedTypeRangeOption>,
+    },
+    /// Base type (SQL definition): `CREATE TYPE name (options)`
+    ///
+    /// Note the lack of `AS` keyword
+    SqlDefinition {
+        options: Vec<UserDefinedTypeSqlDefinitionOption>,
+    },
+    /// When the representation of the type is not specified.
+    /// This is used in `CREATE TYPE <name>;` statements.
+    None,
```

Comment:
Instead of having a special `None` variant, ideally the usage becomes an optional field, would that work here? i.e `representation: Option<UserDefinedTypeRepresentation>`

Reply:
Sure I can switch to that case - should we normalize other analogous enumerations which also have the `None` field, maybe in another PR?

Reply:
Done